### PR TITLE
Add TEST_check

### PR DESCRIPTION
**Pull request by ekasper**

### CODE REVIEW

File: test/handshake_helper.c [hunk 1]
```diff
@@ -14,17 +14,20 @@
 #include <openssl/ssl.h>
 
 #include "handshake_helper.h"
+#include "testutil.h"
 
 HANDSHAKE_RESULT *HANDSHAKE_RESULT_new()
 {
     HANDSHAKE_RESULT *ret;
     ret = OPENSSL_zalloc(sizeof(*ret));
```
Comment (richsalz) [inline test/handshake_helper.c]: blank line missing; maybe merge 21/22.

File: test/handshake_helper.c [hunk 2]
```diff
@@ -544,13 +546,13 @@ static HANDSHAKE_RESULT *do_handshake_internal(
 
     server = SSL_new(server_ctx);
     client = SSL_new(client_ctx);
-    OPENSSL_assert(server != NULL && client != NULL);
+    TEST_check(server != NULL && client != NULL);
```
Comment (richsalz) [inline test/handshake_helper.c]: i prefer not to have multiple conditions in a test.  fine if you want to leave it tho, just style.

File: test/testutil.h [hunk 3]
```diff
@@ -94,3 +96,17 @@ int run_tests(const char *test_prog_name);
...
+/*
+ * For "impossible" conditions such as malloc failures or bugs in test code,
+ * where continuing the test would be meaningless. Note that OPENSSL_assert
+ * is fatal, and is never compiled out.
+ */
+#define TEST_check(condition)                   \
+    do {                                        \
+        if (!(condition)) {                     \
+            ERR_print_errors_fp(stderr);        \
+            OPENSSL_assert(0);                  \
```
Comment (richsalz) [inline test/testutil.h]: maybe ##condition as the argument to the assert?  or whatever it is that stringify's the expression.

**Comment by richsalz** [comment]
+1 no matter what you do with my comments.

File: test/testutil.h (hunk 3 above)
Comment (ekasper) [inline test/testutil.h]: Good idea, done.

**Comment by richsalz** [comment]
+1

**Pull request closed without merging**